strapi/strapi
# docs: api reference

### What does it do?

I have added a bunch of documentation in the Strapi repo. The goal is to document every bit of the API services, modules and classes that can be used in the backend and structure it like the NodeJS API reference.

### Why is it needed?

As someone who discovered the Strapi code not a long time ago, I'm forced to read the codebase each time I want to do something. An API reference is a great tool to do things a lot faster and to avoid lots of headaches.

### How to test it?

Go inside the `docs` folder and run `yarn start`. The docusaurus project will be started and you will be able to see the documentation.

![Full page of Strapi class documentation](https://user-images.githubusercontent.com/3313498/229819282-9fe8e53f-c6cf-497a-b246-3cfa195f8ab1.png)


## ⚠️ Some parts are missing (WIP statement). This is "normal" until the documentation pages are completely finished. This PR is not intended to finish the whole API ref documentation (that will take some time) but to give it a good start.
